### PR TITLE
To Do list: interactive list

### PR DESCRIPTION
**Pull request by ometman**
To Do list interactivity functionality implemented:

- 3 new JS files were added, taskStatus, markCompleted and clearCompleted.
- The files were imported as modules.
- Task status is updated to true or false using a check box with an event listener.
- Implemented a function for "Clear all completed" button using a filter method
- Updates are stored in local storage and status persists on reload.
- Linters checked and passed.

### CODE REVIEW

File: src/modules/addTask.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { TasksClass } from './tasksClass.js';
+// import { retainCheck } from './taskStatus.js';
```
Comment (jerryowusu) [inline src/modules/addTask.js]: As part of the javascript best practices kindly remove this commented code from your codebase. Leaving commented code in your codebase is generally not a good practice. Commented-out code can clutter your codebase, make it harder to read, and can potentially lead to confusion or outdated information. If you need to review a previous version of your code, you can always use git history. :pray:

File: src/modules/markCompleted.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+// mark task complete
+export const markComplete = () => {
+  //  get the elements
+  const taskBoxEls = document.querySelectorAll('.task-select-input');
+  const taskStatus = new TasksClass();
+  let taskBoxValue = false;
+  // get index and checkbox value by iterating
+  taskBoxEls.forEach((theEl, theElIndex) => {
+    // theEl.checked = true;
```
Comment (jerryowusu) [inline src/modules/markCompleted.js]: As part of the javascript best practices kindly remove this commented code from your codebase. Leaving commented code in your codebase is generally not a good practice. Commented-out code can clutter your codebase, make it harder to read, and can potentially lead to confusion or outdated information. If you need to review a previous version of your code, you can always use git history. :pray:

File: src/modules/taskStatus.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { TasksClass } from './tasksClass.js';
+
+//  get task status on checkbox
+export const retainCheck = () => {
+  //  get the elements
+  const taskEls = document.querySelectorAll('.task-select-input');
+  const checkStatus = new TasksClass();
+  // get index and checkbox value by iterating and set checkbox
+  taskEls.forEach((theEl, theElIndex) => {
+    // theEl.checked = true;
```
Comment (jerryowusu) [inline src/modules/taskStatus.js]: As part of the javascript best practices kindly remove this commented code from your codebase. Leaving commented code in your codebase is generally not a good practice. Commented-out code can clutter your codebase, make it harder to read, and can potentially lead to confusion or outdated information. If you need to review a previous version of your code, you can always use git history. :pray:

File: src/modules/tasksClass.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+  taskCompleted = (theElIndex, taskBoxValue) => {
+    const tC = this.taskCollection;
+    tC[theElIndex].taskCompletion = taskBoxValue;
+    localStorage.setItem('taskList', JSON.stringify(this.taskCollection));
+  } // task is marked complete
+
+  clearCompletedTask = (theElIndex) => {
+    const tC = this.taskCollection;
+    // tC.filter((task) => task.taskCompletion === false);
```
Comment (jerryowusu) [inline src/modules/tasksClass.js]: As part of the javascript best practices kindly remove this commented code from your codebase. Leaving commented code in your codebase is generally not a good practice. Commented-out code can clutter your codebase, make it harder to read, and can potentially lead to confusion or outdated information. If you need to review a previous version of your code, you can always use git history. :pray:

File: src/modules/tasksClass.js [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+  displayAllTasks = () => {
+    const tC = JSON.parse(localStorage.getItem('taskList'));
+    // empty container
+    const displayContainer = document.querySelector('#display-container');
+    displayContainer.innerHTML = '';
+    // begin loop
+    tC.forEach((tcTask, tcTaskIndex) => {
+      displayContainer.innerHTML += ` 
+      <div id="${tcTaskIndex}" class="task-container row px-2 ms-0 me-0">
```
Comment (jerryowusu) [inline src/modules/tasksClass.js]: Your project is throwing a TyperError due to the displayAllTasks method as shown below, hence a user is not able to create tasks. Kindly ensure that this error is resolved and the app is function well according to the project requirements.
Kindly remember also that it is part of the project requrements to implement the array.filter() method. :pray: 

Screenshot from 2023-08-19 06-18-46

**Review by jerryowusu — CHANGES_REQUESTED** [review]
Hi @ometman , 

Good job so far!
Your project is looking fantastic so far :star_struck: 

However, there are few changes you need to make in order to move on to the next level :sunglasses:

almost there


## Highlights
Neat and well organized code :+1: 
Great job so far :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/tasksClass.js (hunk 5 above)
Comment (ometman) [reply src/modules/tasksClass.js]: Worked locally. Thanks though. 😄

File: src/modules/taskStatus.js (hunk 3 above)
Comment (ometman) [reply src/modules/taskStatus.js]: I appreciate the reminder.

File: src/modules/markCompleted.js (hunk 2 above)
Comment (ometman) [reply src/modules/markCompleted.js]: I appreciate the reminder.

File: src/modules/addTask.js (hunk 1 above)
Comment (ometman) [reply src/modules/addTask.js]: I appreciate the reminder.

**Review by codepantha — APPROVED** [review]
# STATUS: APPROVED 🟢 ✔️ 

## Hi @ometman 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 🎯 
✔️ Descriptive PR
✔️ Good commit history
✔️ Tasks can be marked as completed by selecting the checkout 📌 
✔️ Tasks are updated in localstorage 📌 
✔️ Completed tasks can be deleted 📌 
✔️ No linter errors 🟢 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ometman** [comment]
Thank you. 😄

**Pull request merged**